### DISCUSSION TITLE
Дополнительная информация в селекте

### DISCUSSION THREAD

**Discussion by M3teora** [totumonline/totum-ru-issues-and-discussions#454] [Как это сделать?] [answered]
Доброго времени суток! 
Столкнулся со следующей задачей. 

Есть JSON вида: 
`[
  {
    "summ": "10.00",
    "kolvo": "1.00",
    "id": "196"
  },
  {
    "itogo": "5.00",
    "kolvo": "1.00",
    "id": "342"
  },
  {
    "itogo": "15.00",
    "kolvo": "2.00",
    "id": "210"
  }
]`

Получается я хочу получить в результате что-то типо списка в одной ячейке напротив каждого заказа, такого вида:
"Товар 1 - "кружка"(id 196) - кол-во 1 - сумма 10.00,
Товар 2 - "нож" (id 342) - кол-во 1 - сумма 5.00,
Товар 3 - "тарелка"(id 210) - кол-во 2 - сумма 15.00" 

Т.е. на выходе у меня будет что-то типо таблицы:

| Дата заказа | Дата отгрузки | Номер заказа | Состав заказа | 

Я не совсем понимаю как в одной строке объединить и значения селектов (вывод названия товара по id) и просто значения из JSON. 
 
 ![totum](https://github.com/totumonline/totum-ru-issues-and-discussions/assets/7896517/de62a41d-cdfa-492a-8b70-455e09cf8a76)

**Comment by ttm-alexeypolunin** [comment, marked as answer]
selectRowListForSelect возвращает структуру и его можно выполнить просто как Код — сделайте в тестовом поле с типом Данные и увидите какой, ну или посмотрите в доках по функции.

Перед отдачей в =: в коде селекта можете перебрать его через listReplace изменяя Title как вам необходимо.

Что бы он считался для каждой строки — включите параметр Индивидуальный расчет блалала.

Это все достаточно сильно поест процессор — проверяйте на объеме, увлекаться не стоит.

**Comment by M3teora** [comment]
Да, я попробовал он добавляет дополнительно ключи "value", "isdel", "title", но в целом же остаются данные такие же как и были на скриншоте выше в столбце "товары в заказе" 
Я попробовал поработать с listReplace, в целом то что мне нужно, однако я не могу понять логику как именно нужно написать код чтобы он перебирал именно значения из json и выбирал из нужной таблицы сопоставляя с нужным ID. 

`
= : listReplace(list: #tovary_v_zakaze; action: "pozicii_v_zakaze" = $sel)
sel: selectList(table: 'goods'; field: 'item_name'; where: 'id' = $list_ids)
list_ids: #tovary_v_zakaze[[pozicii_v_zakaze]]
`
при таком сценарии он выдает что-то типо

 `{
    "itogo": "47.76",
    "kolvo": "1.00",
    "pozicii_v_zakaze": [
      "Отдушка Кашемир - 0.1",
      "Отдушка Дубовый мох и янтарь - 0.03",
      "Деревянный фитиль 17х130 мм - 10 шт",
      "Отдушка Сандаловое дерево - 0.1"
    ]
  }`

ну либо если я буду использовать  вместо selectlist просто select то будет везде одни значения
![telegram-cloud-photo-size-2-5294318422873722346-x](https://github.com/totumonline/totum-ru-issues-and-discussions/assets/7896517/9ddf0e72-ec5b-460d-8f02-4f5f2351f222)

Я не могу добиться именно перебора значений, чтобы тотум сначала взял первую позицию, взял ее id и нашел в списке товаров им, потом вторую и т.д.
Т.е. грубо говоря пройтись циклом for по этим значениям

**Comment by M3teora** [comment]
Мне как-то удалось к чему-то прийти, пока сам не понимаю как это работает, взял за основу какой-то старый код который писал >полугода назад по примеру из демки

В результате получилось вот что-то такое: 

<img width="1157" alt="image" src="https://github.com/totumonline/totum-ru-issues-and-discussions/assets/7896517/53fab0d1-18b6-4c9f-8f3f-4f2c2f16a7db">

<img width="488" alt="image" src="https://github.com/totumonline/totum-ru-issues-and-discussions/assets/7896517/e4b6eb92-97c6-41b7-a759-c8e2a5615658">

Я правда не до конца понимаю как это работает, и наверное есть что-то лишнее)) 
И как это распаковать в удобный вид пока тоже не знаю)
